eigentrust/src/lib.rs

```diff
@@ -456,16 +462,29 @@ impl Client {
 		signed_attestations
 	}
 
-	/// Fetches logs from the contract.
+	/// Fetches "AttestationCreated" event logs from the contract, filtered by domain prefix.
 	pub async fn get_logs(&self) -> Result<Vec<Log>, EigenError> {
-		let filter = Filter::new()
-			.address(self.config.as_address.parse::<Address>().unwrap())
-			.event("AttestationCreated(address,address,bytes32,bytes)")
-			.topic1(Vec::<H256>::new())
-			.topic2(Vec::<H256>::new())
-			.from_block(0);
-
-		self.signer.get_logs(&filter).await.map_err(|e| EigenError::ParsingError(e.to_string()))
+		let as_contract = AttestationStation::new(
+			self.config.as_address.parse::<Address>().unwrap(),
+			self.get_signer(),
+		);
+		let filter = as_contract.attestation_created_filter().filter.from_block(0);
+
+		// Fetch logs matching the filter.
+		let logs = self
+			.signer
+			.get_logs(&filter)
+			.await
+			.map_err(|e| EigenError::ParsingError(e.to_string()))?;
+
+		// Filter logs based on the domain prefix.
+		let filtered_logs: Vec<Log> = logs
+			.iter()
+			.filter(|log| &log.topics[3].as_fixed_bytes()[..DOMAIN_PREFIX_LEN] == DOMAIN_PREFIX)
```

Comment:
What about the domain id? We should also filter based on that

Reply:
Yes! I'm adding that as well